src/main/java/org/jenkinsci/remoting/util/AnonymousClassWarnings.java

```diff
@@ -47,6 +51,8 @@
     private static final Logger LOGGER = Logger.getLogger(AnonymousClassWarnings.class.getName());
     private static final Map<Class<?>, Boolean> checked = new WeakHashMap<>();
 
+    private final static ExecutorService threadPool = Executors.newFixedThreadPool(10, new NamingThreadFactory(new DaemonThreadFactory(), AnonymousClassWarnings.class.getSimpleName()));
```

Comment:
Or 
```suggestion
    private final static ExecutorService threadPool = new AtmostOneThreadExecutor(new NamingThreadFactory(new DaemonThreadFactory(), AnonymousClassWarnings.class.getSimpleName()));
```

but I'm afraid it would slow down startup.

Reply:
> but I'm afraid it would slow down startup.

as the checks are asynchronous why would this slow down startup (am I missing something?)

The queue for the tasks may grow quite large - but if the issue was this exhausting Threads previously then we would have been allocating stack / native for the threads.  It may mean that an individual task is in memory longer but that would be heap.

Reply:
I thought it could be asynchronous in some cases, but synchronous in others.